backend/app/services/attachment/parser.py

```
@@ -0,0 +1,383 @@
+        # Try common encodings in order of preference
+        encodings_to_try = ["utf-8", "utf-8-sig", "gbk", "gb2312", "gb18030", "latin-1"]
+
+        # First, try to detect encoding using chardet
+        try:
+            detected = chardet.detect(binary_data)
+            detected_encoding = detected.get("encoding")
+            if detected_encoding and detected_encoding.lower() not in ["ascii", "charmap"]:
+                # Insert detected encoding at the beginning if it's valid
+                encodings_to_try.insert(0, detected_encoding)
+        except Exception:
+            pass  # Ignore chardet errors
```

Ruff (0.14.7)

`try`-`except`-`pass` detected, consider logging the exception

(S110)